src/Features/CSharp/Portable/CodeRefactorings/ExtractClass/CSharpExtractClassCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Composition;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.CodeRefactorings.PullMemberUp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.ExtractClass;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeRefactorings.ExtractClass
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = nameof(PredefinedCodeRefactoringProviderNames.ExtractClass)), Shared]
+    internal class CSharpExtractClassCodeRefactoringProvider : AbstractExtractClassRefactoringProvider
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public CSharpExtractClassCodeRefactoringProvider()
+        {
+        }
+
+        /// <summary>
+        /// Test purpose only.
+        /// </summary>
+        [SuppressMessage("RoslynDiagnosticsReliability", "RS0034:Exported parts should have [ImportingConstructor]", Justification = "Used incorrectly by tests")]
+        internal CSharpExtractClassCodeRefactoringProvider(IExtractClassOptionsService optionsService)
+            : base(optionsService)
+        {
+        }
+
+        protected override async Task<SyntaxNode?> GetSelectedClassDeclarationAsync(CodeRefactoringContext context)
+        {
+            var relaventNodes = await context.GetRelevantNodesAsync<ClassDeclarationSyntax>().ConfigureAwait(false);
+            return relaventNodes.FirstOrDefault();
+        }
+
+        protected override Task<SyntaxNode?> GetSelectedNodeAsync(CodeRefactoringContext context) => NodeSelectionHelpers.GetSelectedDeclarationOrVariableAsync(context);
```

Comment:
```suggestion
        protected override Task<SyntaxNode?> GetSelectedNodeAsync(CodeRefactoringContext context)
            => NodeSelectionHelpers.GetSelectedDeclarationOrVariableAsync(context);
```

the reason for this is htat, by default, when a method is single-line it just looks abstract to me.  and i use enough cognition to just scan the signature.  by being on multiple lines, i can more easily make out hte sig and then go "ok, now let's look at the body".